crates/biome_module_graph/src/jsdoc_comment.rs

```diff
@@ -0,0 +1,118 @@
+use std::ops::Deref;
+
+use biome_js_syntax::{JsSyntaxNode, JsSyntaxToken};
+use biome_rowan::TriviaPieceKind;
+
+/// Represents a normalised JSDoc comment.
+///
+/// Comments are trimmed and normalised to remove the trivia of the comment.
+///
+/// ## Example
+///
+/// Assuming the following JSDoc comment:
+///
+/// ```ts
+/// /**
+///  * Magic constant of fooness.
+///  *
+///  * For if you want more ways to write 1.
+///  */
+/// export const FOO = 1;
+/// ```
+///
+/// The normalised representation will become:
+/// `"Magic constant of fooness.\n\nFor if you want more ways to write 1."`.
+///    
+/// See https://jsdoc.app/ for the JSDoc reference.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct JsdocComment(String);
+
+impl JsdocComment {
+    /// Creates a normalised JSDoc comment from the given comment `text`.
+    pub fn from_comment_text(text: &str) -> Self {
+        debug_assert!(text.starts_with("/**") && text.ends_with("*/"));
+
+        let mut result = text[3..text.len() - 2]
+            .lines()
+            .map(|line| {
+                let trimmed = line.trim();
+                trimmed.strip_prefix('*').map_or(trimmed, str::trim_start)
+            })
+            .fold(String::new(), |mut result, line| {
+                if !result.is_empty() {
+                    result.push('\n');
+                }
+                result.push_str(line);
+                result
+            });
+
+        // Trim trailing newlines.
+        while result.ends_with('\n') {
```

Comment:
We should consider `\r` and `\r\n` too

Reply:
There can't be a `\r`, because we pushed the `\n` ourselves a few lines above and the lines were split using `.lines()`, which already handles different newline conventions.